[dynamo] [2/3] added add_new_gb_type functionality

Sidharth123-cpu opened the pull request:
The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />


Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #154886
* #154738

Commit:
[dynamo] added add_new_gb_type functionality

[ghstack-poisoned]

Commit:
Update on "[dynamo] [2/N] added add_new_gb_type functionality"




[ghstack-poisoned]

williamwen42 commented on tools/dynamo/gb_id_mapping.py:
```diff
@@ -137,6 +159,39 @@ def find_unimplemented_v2_calls(dynamo_dir):
+
+    save_registry(reg, registry_path)
+    print(f"Added {gb_type} to registry with ID {gb_id}")
+    return True
```
Are we using the return for anything?

williamwen42 commented on tools/dynamo/gb_id_mapping.py:
```diff
@@ -137,6 +159,39 @@ def find_unimplemented_v2_calls(dynamo_dir):
     return results
 
 
+def cmd_add_new_gb_type(gb_type, file_path, registry_path):
```
Add a docstring

Commit:
Update on "[dynamo] [2/N] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

Commit:
Update on "[dynamo] [2/N] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

Commit:
Update on "[dynamo] [2/N] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

Commit:
Update on "[dynamo] [2/3] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

Commit:
Update on "[dynamo] [2/3] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

Commit:
Update on "[dynamo] [2/3] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

williamwen42 commented on tools/dynamo/gb_id_mapping.py:
```diff
@@ -8,42 +8,60 @@
-        s = s.replace("\\", "")
+        # Replace adjacent quoted strings with a space (e.g., " "" -> " ")
         s = re.sub(r'" "', " ", s)
+        # Remove any curly brace expressions used in f-strings (e.g., {variable})
```
I think we should keep the curly braces so the user is aware of what the string is supposed to show.

williamwen42 commented on tools/dynamo/gb_id_mapping.py:
```diff
@@ -8,42 +8,60 @@
+    """
+    Normalizes string literals by removing formatting artifacts and escape sequences.
+    Handles f-strings, quotes, newlines, and other syntax elements for cleaner output.
+    """
```
Is it possible to reduce the number of regex substitutions here? e.g. there should be some builtin methods to remove escaped sequences.

Sidharth123-cpu commented on tools/dynamo/gb_id_mapping.py:
```diff
@@ -8,42 +8,60 @@
     if isinstance(s, str):
+        # Convert f-string prefix to regular string prefix (e.g., f"hello" -> "hello")
         s = re.sub(r'^f["\']', r'"', s)
+        # Replace quoted strings with f-prefix in the middle with a space (e.g., " f"" -> " ")
```
This regex handles cases where an f-string appears in the middle of concatenated strings, like `"text" f"variable {x}" "more text"`. There were a few cases where this occured.

Commit:
Update on "[dynamo] [2/3] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

williamwen42 commented on tools/dynamo/gb_id_mapping.py:
```diff
@@ -151,39 +207,57 @@ def create_registry(dynamo_dir, registry_path):
 
     with open(registry_path, "w") as f:
         json.dump(registry, f, indent=2)
+    return True
```
When would this function return False?

Commit:
Update on "[dynamo] [2/3] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

Commit:
Update on "[dynamo] [2/3] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

Commit:
Update on "[dynamo] [2/3] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

williamwen42 commented on tools/dynamo/gb_id_mapping.py:
```diff
@@ -154,36 +210,53 @@ def create_registry(dynamo_dir, registry_path):
-    create_registry(args.dynamo_dir, args.registry_path)
+    if args.command == "create":
+        success = create_registry(args.dynamo_dir, args.registry_path)
+        if not success:
```
create_registry returns `None` right now, so we would always exit with status 1.

Commit:
Update on "[dynamo] [2/3] added add_new_gb_type functionality"

The user can now use the terminal to update the registry whenever they create a new unimplemented_v2() callsite.
Terminal command template: python [path to gb_id_mapping.py] add "new_gb_type" [path to file where user added callsite]
Before the user added a new gb_type:
<img width="619" alt="Screenshot 2025-06-02 at 1 33 54 PM" src="https://github.com/user-attachments/assets/7258cab1-a184-4200-9d56-7b21d243d6d8" />
After the user added a new gb_type:
<img width="366" alt="Screenshot 2025-06-02 at 1 34 47 PM" src="https://github.com/user-attachments/assets/5c383e94-268c-4f6d-9111-7b18c856222e" />





[ghstack-poisoned]

Sidharth123-cpu commented:
@pytorchbot merge -f "intentional parameter name change"
